### DISCUSSION TITLE
contiki-ng/examples/libs/shell/ too minimal

### DISCUSSION THREAD

**Discussion by NestorESA** [contiki-ng/contiki-ng#2407] [General]
I believe there is supposed to be code here to show an example of the use of shell, I couldnt see anything, can you tell me where can I find a proper example?
in the website it only displays this:
This is a minimal example for the shell.

Thats way tooo minimal, there is no example in there.

**Comment by arurke** [comment]
The README is a bit too minimal indeed. The only thing needed to enable shell is to add `MODULES += os/services/shell` in the Makefile, which is what the example is doing. Have a look at https://docs.contiki-ng.org/en/develop/doc/tutorials/Shell.html for more information.

**Reply by NestorESA** [reply]
My bad, I didnt explain enough, I know that I have to add that to the the makefile but to I couldn't find much information on what functions should be used to send a command to the shell from the C file. I tried several ways but it never recognized the functions, it wasn't a header problem because I put the correspondent headers.

**Reply by arurke** [reply]
The shell is designed to receive and send via the serial port of the device (ref. the link to the tutorial). If you want to have a C file which does what some shell command does, I would suggest investigating shell-commands.c/.h to learn what each command is doing when executed.
